### PR TITLE
API-9105 | Service History Gras Integration

### PR DESCRIPTION
**Pull request by lukemarkwordtlibertyits**
# Description

Added filtering for Title 32 Reserve periods.

Why: The base get_military_service_episodes_v2 service includes potentially non-active guard service obligations like training, rather than actual service history.

As such, we need more advanced filtering to return the correct service periods for benefit eligibility.

Service History will now be a combination of the Military Service Episodes and Guard Reserve Service Periods responses, with the filtering below.

```
1. get_military_service_episodes_v2, personnel_category_type_code
  - If A, then active and should be included

2. get_guard_reserve_service_periods, trainingIndicatorCode ( this is completely separate logic from military service episodes )
      - If Y, then should be filtered out
      - If N, then more filtering is required
        3. get_guard_reserve_service_periods, guardReservePeriodStatuteCode
           - If [J,N,P,Q,Z] then title 32 and should be filtered out
           - Otherwise, then active duty guard period and should be included
```

## JIRA

https://vajira.max.gov/browse/API-9105

## Quick Start

Because of the the GRAS integration logic, some service episodes have been filtered out and some reserve periods have been added to test user responses.

run the following command against master and then against this branch to see the difference.

```
curl http://localhost:8080/v0/service_history/1012832025V743496
```

see 
[vv_java_dif.log](https://github.com/department-of-veterans-affairs/lighthouse-veteran-verification/files/7172156/vv_java_dif.log) for all changes to responses.

## Related PRs

https://github.com/department-of-veterans-affairs/lighthouse-veteran-verification-deployment/pull/27

## Blocking PRs

**This PR has been merged so it is no longer blocking**
https://github.com/department-of-veterans-affairs/lighthouse-enterprise-military-information-service/pull/6

### CODE REVIEW

**Comment by dbrown-va** [comment]
Some minor conflicts in the pom from another PR.

**Comment by esubocz-va** [comment]
It looks like this is an existing issue in master that we missed, but the response is coming through with camel case field names rather than with underscores. (i.e. `firstName` vs what we're looking for with `first_name`)

File: veteran-verification/src/main/resources/application.properties [hunk 1]
```diff
@@ -13,5 +13,5 @@ management.endpoints.web.exposure.include=health,info
 
 logging.stack-trace-filter=java.base,java.lang.reflect.Method, org.apache.coyote, org.apache.catalina, org.apache.tomcat, org.springframework.aop, org.springframework.security, org.springframework.transaction, org.springframework.web, org.springframework.cglib, org.springframework.validation.beanvalidation, org.springframework.boot.actuate, javax.servlet.http,  reactor.core, reactor.ipc, sun.reflect, net.sf.cglib, ByCGLIB, io.netty
 logging.pattern.console=%clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr(%5p) %clr(${PID: }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n%wEx{full,${logging.stack-trace-filter}}}
-spring.jackson.default-property-inclusion=non_empty
```
Comment (lukemarkwordtlibertyits) [inline veteran-verification/src/main/resources/application.properties:16]: to stay consistent to previous functionality, we will include empty arrays. we can remove this later if we would like.

File: veteran-verification/src/main/resources/application.properties [hunk 2]
```diff
@@ -13,5 +13,6 @@ management.endpoints.web.exposure.include=health,info
...
+spring.jackson.default-property-inclusion=non_null
```
Comment (lukemarkwordtlibertyits) [inline veteran-verification/src/main/resources/application.properties:16]: This is consistent with previous java VV functionality; however vets-api VV includes null fields. I can change it to be consistent with vets-api if desired.

File: veteran-verification/src/main/java/gov/va/api/lighthouse/veteranverification/service/controller/servicehistory/VeteranServiceHistoryTransformer.java [hunk 3]
```diff
@@ -206,15 +136,136 @@ public ServiceHistoryResponse serviceHistoryTransformer() {
...
               .id(serviceHistoryId)
               .build());
-
       unusedDeployments.removeAll(attributes.deployments);
     }
-    return ServiceHistoryResponse.builder().data(episodes.stream().toList()).build();
+    return new ArrayList<>(episodes);
+  }
+
+  private List<ServiceHistoryResponse.ServiceHistoryEpisode> getNonTitle32ReservePeriods() {
```
Comment (dbrown-va) [inline veteran-verification/src/main/java/gov/va/api/lighthouse/veteranverification/service/controller/servicehistory/VeteranServiceHistoryTransformer.java:149]: Recommend both of these methods have comments with explanations of the filtering that is going on.

File: veteran-verification/src/main/java/gov/va/api/lighthouse/veteranverification/service/controller/servicehistory/VeteranServiceHistoryTransformer.java [hunk 4]
```diff
@@ -23,14 +30,34 @@
...
+      Map.ofEntries(
+          new AbstractMap.SimpleEntry<>("O", "NOAA"),
+          new AbstractMap.SimpleEntry<>("H", "Public Health Service"),
+          new AbstractMap.SimpleEntry<>("A", "Army"),
+          new AbstractMap.SimpleEntry<>("C", "Coast Guard"),
+          new AbstractMap.SimpleEntry<>("F", "Air Force"),
+          new AbstractMap.SimpleEntry<>("M", "Marine Corps"),
+          new AbstractMap.SimpleEntry<>("N", "Navy"));
+
+  private final String[] thirtyTwoStatuteCodes = {"J", "N", "P", "Q", "Z"};
```
Comment (dbrown-va) [inline veteran-verification/src/main/java/gov/va/api/lighthouse/veteranverification/service/controller/servicehistory/VeteranServiceHistoryTransformer.java:49]: Recommend renaming to `title32StatuteCodes` as that is the specific term in the US Code.

File: veteran-verification/src/main/resources/application.properties (hunk 2 above)
Comment (dbrown-va) [reply veteran-verification/src/main/resources/application.properties:16]: Consistency with vets-api will make our cutover the simplest from a regression testing perspective.  I would lean towards that.

**Review by esubocz-va — APPROVED** [review]
Functionality looks good to me, and all feedback appears to be addressed.

**Pull request merged**